MediaBrowser.Api/Playback/BaseStreamingService.cs

```diff
@@ -779,12 +755,12 @@ protected async Task<StreamState> GetState(StreamRequest request, CancellationTo
                     var mediaSources = (await MediaSourceManager.GetPlayackMediaSources(LibraryManager.GetItemById(request.Id), null, false, false, cancellationToken).ConfigureAwait(false)).ToList();
 
                     mediaSource = string.IsNullOrEmpty(request.MediaSourceId)
-                       ? mediaSources.First()
-                       : mediaSources.FirstOrDefault(i => string.Equals(i.Id, request.MediaSourceId));
+                       ? mediaSources[0]
+                       : mediaSources.Find(i => string.Equals(i.Id, request.MediaSourceId));
```

Comment:
why replace `FirstOrDefault` with `Find`?

Reply:
`FirstOrDefault` is a generic function from LINQ.
`Find` is a function specifically for `Lists`.
In other words: `Find` is faster 🤷‍♂️